src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -1615,31 +1615,14 @@ private void getCallInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_RESO
 
                 pResult->nullInstanceCheck = false;
             }
-            else if ((flags & CORINFO_CALLINFO_FLAGS.CORINFO_CALLINFO_LDFTN) == 0
-                // Canonically-equivalent types have the same vtable layout. Check the canonical form.
-                // We don't want to accidentally ask about Foo<object, __Canon> that may or may not
-                // be available to ask vtable questions about.
-                // This can happen in inlining that the scanner didn't expect.
-                && _compilation.HasFixedSlotVTable(targetMethod.OwningType.ConvertToCanonForm(CanonicalFormKind.Specific)))
+            else if ((flags & CORINFO_CALLINFO_FLAGS.CORINFO_CALLINFO_LDFTN) == 0)
```

Comment:
Is there a reason for suppressing VTABLE slot for CORINFO_VIRTUALCALL_VTABLE? We should be able to take it to fetch the virtual delegate target too (it is probably not super common).

Reply:
I'll look into this once we get runtime tests testing in order. As you say, it's not super common - I think we'd only exercise the results of this when loading addresses of virtual methods in situations that are not verifiable-delegate-sequence. Those tests are in the runtime test tree.